src/client/datascience/serviceRegistry.ts

```diff
@@ -203,6 +207,14 @@ export function registerTypes(serviceManager: IServiceManager, inNotebookApiExpe
     serviceManager.addSingletonInstance<boolean>(UseVSCodeNotebookEditorApi, useVSCodeNotebookAPI);
     serviceManager.addSingletonInstance<number>(DataScienceStartupTime, Date.now());
 
+    // This will ensure all subsequent telemetry will get the context of whether it is a custom/native/old notebook editor.
+    // This is temporary, and once we ship native editor this needs to be removed.
+    setSharedProperty('ds_notebookeditor', useVSCodeNotebookAPI ? 'native' : UseCustomEditorApi ? 'custom' : 'old');

```

Comment:
User cannot use two at a time, they are either in native or not (easier to send at startup).